packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -0,0 +1,384 @@
+
+  const apps = config.apps || [];
+  // convert the apps to projects
+  const projectMap = apps
+    .map((app: AppConfig, idx: number) => {
+      const name = app.name || `app${idx + 1}`;
+      const outDir = app.outDir || defaults.outDir;
+      const projectRoot = app.root || defaults.appRoot;
+
+      const project: JsonObject = {
+        root: projectRoot,
+        projectType: 'application',
+        cli: {},
+        schematics: {},
+      };
+
+      const extraEntryMapper = (extraEntry: string | JsonObject) => {
+        let entry: JsonObject;
+        if (typeof extraEntry === 'string') {
+          entry = { input: extraEntry };
+        } else {
+          entry = extraEntry;
+        }
+        entry.input = join(projectRoot as Path, <string> entry.input || '');
+
+        return entry;
+      };
+
+      const architect: JsonObject = {};
+      project.architect = architect;
+
+        // Browser target
+      const buildOptions: JsonObject = {
+        // Make outputPath relative to root.
+        outputPath: relative(normalize('/' + project.root), normalize('/' + outDir)),
+        index: projectRoot + '/' + app.index || defaults.index,
+        main: projectRoot + '/' + app.main || defaults.main,
+        polyfills: projectRoot + '/' + app.polyfills || defaults.polyfills,
+        tsConfig: projectRoot + '/' + app.tsconfig || defaults.tsConfig,
+      };
+
+      buildOptions.assets = (app.assets || []).map((asset: string | JsonObject) =>
+        typeof asset === 'string'
+          ? { glob: projectRoot + '/' + asset }
+          : projectRoot + '/' + asset);
+      buildOptions.styles = (app.styles || []).map(extraEntryMapper);
+      buildOptions.scripts = (app.scripts || []).map(extraEntryMapper);
+      architect.build = {
+        builder: `${builderPackage}:browser`,
+        options: buildOptions,
+        configurations: {
+          production: {
+            optimization: true,
+            outputHashing: 'all',
+            sourceMap: false,
+            extractCss: true,
+            namedChunks: false,
+            aot: true,
+            extractLicenses: true,
+            vendorChunk: false,
+            buildOptimizer: true,
+          },
+        },
+      };
+
+      // Serve target
+      const serveOptions: JsonObject = {
+        browserTarget: `${name}:build`,
+      };
+      architect.serve = {
+        builder: `${builderPackage}:dev-server`,
+        options: serveOptions,
+        configurations: {
+          production: {
+            browserTarget: `${name}:build`,
+          },
+        },
+      };
+
+      // Extract target
+      const extractI18nOptions: JsonObject = { browserTarget: `${name}:build` };
+      architect['extract-i18n'] = {
+        builder: `${builderPackage}:extract-i18n`,
+        options: extractI18nOptions,
+      };
+
+        // TODO: decide how to handle config.test or config.test.karma are not defined
+      const karmaConfig = config.test && config.test.karma
```

Comment:
If they are not defined, the old CLI could not have used them at all. So not using them in the workspace is the right approach.